### PR TITLE
feat: [AXIMST-17] Course unit - make Problem button functional

### PR DESCRIPTION
**Pull request by ihor-romaniuk**
**Description:** make Problem button functional

**Youtrack:** https://youtrack.raccoongang.com/issue/AXIMST-17

### CODE REVIEW

File: src/course-unit/add-component/AddComponent.jsx [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
 
+  const handleCreateNewXblock = (type) => () => {
+    switch (type) {
+    case COMPONENT_ICON_TYPES.discussion:
+    case COMPONENT_ICON_TYPES.dragAndDrop:
+      handleCreateNewCourseXblock({ type, parentLocator: blockId });
+      break;
+    case COMPONENT_ICON_TYPES.problem:
+      handleCreateNewCourseXblock({ type, parentLocator: blockId }, ({ courseKey, locator }) => {
+        navigate(`/course/${courseKey}/editor/problem/${locator}`);
```
Comment (ihor-romaniuk) [inline src/course-unit/add-component/AddComponent.jsx:25]: [important] cover by util for path transform (adding `course-authoring/` at the beginning)

File: src/utils.js [hunk 2]
```diff
@@ -256,3 +256,25 @@ export const isValidDate = (date) => {
...
+
+/**
+ * Create a correct inner path depend on config PUBLIC_PATH.
+ * @param {string} checkPath - the internal route path that is validated
+ * @returns {string} - the correct internal route path
+ */
+export const createCorrectInternalRoute = (checkPath) => {
+  let basePath = getPath(getConfig().PUBLIC_PATH);
+
+  // Remove the trailing '/'
```
Comment (PKulkoRaccoonGang) [inline src/utils.js:268]: [nit] I think this is an unnecessary comment

File: src/utils.js [hunk 3]
```diff
@@ -256,3 +256,25 @@ export const isValidDate = (date) => {
...
+  if (basePath.endsWith('/')) {
+    basePath = basePath.slice(0, -1);
+  }
+
+  // Check if the check path starts with the base path
```
Comment (PKulkoRaccoonGang) [inline src/utils.js:273]: and this also

File: src/utils.js [hunk 4]
```diff
@@ -256,3 +256,25 @@ export const isValidDate = (date) => {
...
+  if (!checkPath.startsWith(basePath)) {
+    return `${basePath}${checkPath}`;
+  }
+
+  // Return path as is
```
Comment (PKulkoRaccoonGang) [inline src/utils.js:278]: and this
@ihor-romaniuk what do you think?

File: src/utils.js (hunk 2 above)
Comment (ihor-romaniuk) [reply src/utils.js]: Agree, removed

File: src/utils.js (hunk 3 above)
Comment (ihor-romaniuk) [reply src/utils.js]: Removed

File: src/utils.js (hunk 4 above)
Comment (ihor-romaniuk) [reply src/utils.js]: Removed

**Review by monteri — APPROVED** [review]

**Pull request merged**